csharp/ql/lib/semmle/code/csharp/security/dataflow/UrlRedirectQuery.qll

```diff
@@ -139,6 +139,75 @@ class LocalUrlSanitizer extends Sanitizer {
   LocalUrlSanitizer() { this = DataFlow::BarrierGuard<isLocalUrlSanitizer/3>::getABarrierNode() }
 }
 
+/**
+ * A argument to a call to `List.Contains()` that is a sanitizer for URL redirects.
+ */
+private predicate isContainsUrlSanitizer(Guard guard, Expr e, AbstractValue v) {
+  exists(MethodCall method | method = guard |
+    exists(Method m | m = method.getTarget() |
+      m.hasName("Contains") and
```

Comment:
This is any method with the name `Contains`. Should we narrow this to `Contains` methods defined on specific classes or interfaces? Did you intend to only have this to apply to the `Contains` method on `ICollection<T>`?
Maybe we should consider to leave it as is. There are lots of collection like interfaces and classes that have a `Contains` method. We can narrow this in case we get false positive reports.

Reply:
I thought a little about this, and I agree that we should keep it wide, so we also find `Contains` method in other interfaces.  

I'll add a comment about it.